honeybadger/payload.py

```diff
@@ -94,15 +95,37 @@ def create_payload(exception, exc_traceback=None, config=None, context={}):
     if exc_traceback is None:
         exc_traceback = sys.exc_info()[2]
 
-    payload = default_plugin_manager.generate_payload(config, context)
+    request_payload = default_plugin_manager.generate_payload(config, context)
 
-    return {
+    payload =  {
         'notifier': {
             'name': "Honeybadger for Python",
             'url': "https://github.com/honeybadger-io/honeybadger-python",
             'version': __version__
         },
         'error':  error_payload(exception, exc_traceback, config),
         'server': server_payload(config),
-        'request': payload,
+        'request': request_payload,
+        'details': {}
     }
+
+    if config.is_aws_lambda_environment:
+        AWS_ENV_MAP = [
+            '_HANDLER',
+            'AWS_REGION',
+            'AWS_EXECUTION_ENV',
+            'AWS_LAMBDA_FUNCTION_NAME',
+            'AWS_LAMBDA_LOG_GROUP_NAME',
+            'AWS_LAMBDA_LOG_STREAM_NAME',
+            'AWS_LAMBDA_FUNCTION_VERSION',
+            'AWS_LAMBDA_FUNCTION_MEMORY_SIZE',
+            ]
+
+        lambda_details = {key:os.environ.get(key, None) for key in AWS_ENV_MAP}
+        payload['details']['Lambda Details'] = lambda_details
```

Comment:
@Kelvin4664 can you normalize the key names like [in the Ruby gem](https://github.com/honeybadger-io/honeybadger-ruby/blob/3c8c4ff191ba39ca77305d129b004826a08bb919/lib/honeybadger/util/lambda.rb#L5)?

Here's the mapping in Ruby for reference, where it's `"environment variable name" => "key name for the details object"`

```ruby
AWS_ENV_MAP = {
  "_HANDLER" => "handler",
  "AWS_REGION" => "region",
  "AWS_EXECUTION_ENV" => "runtime",
  "AWS_LAMBDA_FUNCTION_NAME" => "function",
  "AWS_LAMBDA_FUNCTION_MEMORY_SIZE" => "memory",
  "AWS_LAMBDA_FUNCTION_VERSION" => "version",
  "AWS_LAMBDA_LOG_GROUP_NAME" => "log_group",
  "AWS_LAMBDA_LOG_STREAM_NAME" => "log_name"
}
```